packages/sandpack-core/src/resolver/resolver.test.ts

```diff
@@ -503,6 +503,61 @@ describe('resolve', () => {
         '/node_modules/package-exports/src/utils/test/index.js'
       );
     });
+
+    it('should ignore the closest package.json and aliases in it when exports are available in the root', () => {
+      const resolved = resolveSync('@emotion/react/jsx-runtime', {
+        filename: '/foo.js',
+        extensions: ['.ts', '.tsx', '.js', '.jsx'],
+        isFile,
+        readFile,
+      });
+      expect(resolved).toBe(
+        '/node_modules/@emotion/react/jsx-runtime/dist/emotion-react-jsx-runtime.cjs.js'
+      );
+    });
+
+    it('should always use exports from the root of the package and not from the closest package.json', () => {
+      const resolved = resolveSync('exports-from-root/nested', {
+        filename: '/foo.js',
+        extensions: ['.ts', '.tsx', '.js', '.jsx'],
+        isFile,
+        readFile,
+      });
+      // this should specifically not resolve to `/node_modules/exports-from-root/nested/file.js`
+      // package.json#exports should only be used from the root of the package
+      expect(resolved).toBe('/node_modules/exports-from-root/file.js');
+    });
+
+    it('should handle conditional root exports', () => {
+      const resolved = resolveSync('its-fine', {
+        filename: '/foo.js',
+        extensions: ['.ts', '.tsx', '.js', '.jsx'],
+        isFile,
+        readFile,
+      });
+      expect(resolved).toBe('/node_modules/its-fine/out/index.cjs');
+    });
+
+    it('resolve fflate correctly', () => {
+      const resolved = resolveSync('fflate', {
+        filename: '/node_modules/its-fine/out/index.cjs',
+        extensions: ['.ts', '.tsx', '.js', '.jsx'],
+        isFile,
+        readFile,
+      });
+      expect(resolved).toBe('/node_modules/fflate/lib/index.cjs');
+    });
+
+    // exports would return invalid value, so aliasing has to happen afterwards to fix this
```

Comment:
This comment doesn't sound completely right but maybe I'm misreading this. This test case tests if an internal request (within a package) is able to be resolved through the alias field (`package.json#browser`). In such a situation - the `exports` shouldn't even be examined (when the target specifier is a relative path).

Reply:
I'll rephrase it a bit, because it is as you mention it should just be able to respect browser field, has nothing to do with exports.

Currently it also still behaves incorrectly, because now it will read exports even when doing relative requests

Reply:
But I'll make this a two step approach in a follow-up.

If it's a node_module request => use/read exports and main fields
Afterwards and relative paths => read/use aliases

Reply:
> Afterwards and relative paths => read/use aliases

just note here that `import 'pkg'` from within `pkg` should still resolve `exports`. But that might just work out of the box - it most likely only depends on what `node_modules` paths you are looking in for "external requests"

Reply:
Yes, I think we need to add some flag to options like `relativeRequest: boolean` that is defined based on the first character of the original import request being anything but a letter or `@` symbol, pointing to it being an alias/relative import

But that requires quite a bit of refactoring as this also means splitting our pkg.json parsing in two and adding a bunch more tests, will do that in a follow-up, don't wanna make big steps here.

Reply:
It also reminds me of something... you might not handle `package.json#imports` at all. Those are for "internal" requests, the structure of this field is basically the same as `package.json#exports` but all specifiers have to start with `#`.

Reply:
The standalone resolver does support this and hoping to migrate over to using that one later, but it's really dangerous to introduce any big change in this repo

https://github.com/codesandbox/sandpack-resolver/blob/main/src/utils/imports.ts